### DISCUSSION TITLE
Battery levels

### DISCUSSION THREAD

**Discussion by GeorgeCastanza** [andesse/hue-clip-api.node-red-flows#34] [Q&A] [answered]
Is there any way to check battery levels from within Node Red for the Hue battery devices?  I dont seem to be able to check within the Hue app.

**Comment by andesse** [comment]
@GeorgeCastanza Hi, I made a flow for you and also published it. 

[https://github.com/andesse/hue-clip-api.node-red-flows/blob/main/flows/power_state.json](url)

It will show the endpoints including battery levels and sorts it using a switch, so low battery level devices can easily determined. In further steps you might want to replace the ID with a real Device Names (like in your app) using change nodes. But it’s up to you. I could imagine to send the levels also to a dashboard node, to have an overview. It’s also up to you. 

Hope this helps. Greetings
Andreas

**Comment by andesse** [comment]
@GeorgeCastanza 
I’ve just built a flow where it should be able to see all battery states with the Name of the device. I’ll add it probably tomorrow to the repository. There are still some things to do.

**Comment by andesse** [comment, marked as answer]
@GeorgeCastanza can you try if this flow works for you?
I have a friend that has problems with the included flow variables in the subflow. 
this should return clear description name and the battery level. Please confirm. 

`[{"id":"96970f5963ff2da7","type":"subflow","name":"Battery_Levels","info":"","category":"","in":[{"x":140,"y":260,"wires":[{"id":"0f15762b3078c21c"},{"id":"f481a261c4321a39"},{"id":"10e848d2fb7a5deb"}]}],"out":[{"x":1320,"y":760,"wires":[{"id":"b7ab41aa67eec789","port":0},{"id":"5a65d328a2ee213c","port":0},{"id":"807882b3003ba754","port":0},{"id":"c94227ec66636aa2","port":0},{"id":"c27860d7b55163a4","port":0},{"id":"0c590f30582e2e0d","port":0},{"id":"8f06f48a01d0abf3","port":0},{"id":"ace6c6dbd6762da8","port":0},{"id":"be9175d1fcfc0290","port":0},{"id":"7473dfbd102357a2","port":0},{"id":"81ec8c877ccc458b","port":0},{"id":"e1184fa9fc1b2e75","port":0},{"id":"0e11a39dd27e23b3","port":0},{"id":"7f277676409b85ce","port":0},{"id":"04eaeab9bb46cf34","port":0},{"id":"3bd0cf0250b7a9d6","port":0}]}],"env":[{"name":"API-Key","type":"str","value":"123456abcdef"},{"name":"IP-Adress","type":"str","value":"192.168.xxx.xxx"}],"meta":{},"color":"#12d1aD","icon":"node-red/db.svg"},{"id":"8dac534a5ce3b10d","type":"http request","z":"96970f5963ff2da7","name":"https request","method":"use","ret":"obj","paytoqs":"ignore","url":"","tls":"4bc31af9bd03bbdc","persist":false,"proxy":"","insecureHTTPParser":false,"authType":"","senderr":false,"headers":[],"x":550,"y":320,"wires":[["7e9df7b4ae3c0b53","07a988dfef5c46b0"]]},{"id":"1735ae4db805ff41","type":"change","z":"96970f5963ff2da7","name":"power_state","rules":[{"t":"delete","p":"payload","pt":"msg"},{"t":"delete","p":"topic","pt":"msg"},{"t":"set","p":"endpoint","pt":"msg","to":"device_power","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":330,"y":220,"wires":[["2d18cae18e6360ba"]]},{"id":"09f8869bf26266f4","type":"change","z":"96970f5963ff2da7","name":"payload.data","rules":[{"t":"move","p":"payload.data","pt":"msg","to":"payload","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":550,"y":400,"wires":[["d8ae5c6eb43a9514"]]},{"id":"ed8bbf569aa2f8da","type":"function","z":"96970f5963ff2da7","name":"Devices Button and Motion","func":"// NOTE: Devices have multipe rtypes in the services\n// This script grabs the first one, so a motion sensor may show\n// up as \"temperature\" as it also does temp. don't worry\n// this script is more to get the name of the devices to help\n// with the button/motion scripts.\nvar response = [];\nvar devices = {};\nmsg.payload.data.forEach(function(item){\n    try {\n        if(item.metadata.archetype == \"unknown_archetype\") {\n            var pair = {\n                \"name\": item.metadata.name,\n                \"id\": item.id,\n                \"rid\": item.services[0].rid,\n                \"rtype\": item.services[0].rtype,\n                \"power\": item.services[1].rid\n            };\n            devices[item.id] = item.metadata.name;\n            response.push(pair);\n        }\n    } catch(err) {\n        node.warn(err);\n    }\n});\n\nreturn {\n    \"payload\": response,\n    \"topic\": \"devices name/rid\",\n    \"keys\": devices\n};","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":920,"y":280,"wires":[["b90ab8a670739d83"]]},{"id":"f481a261c4321a39","type":"change","z":"96970f5963ff2da7","name":"device","rules":[{"t":"delete","p":"payload","pt":"msg"},{"t":"delete","p":"topic","pt":"msg"},{"t":"set","p":"endpoint","pt":"msg","to":"device","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":310,"y":340,"wires":[["1c20b4f9de1af513"]]},{"id":"c94227ec66636aa2","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.3.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.3.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":580,"wires":[[]]},{"id":"3bd0cf0250b7a9d6","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.4.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.4.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":620,"wires":[[]]},{"id":"04eaeab9bb46cf34","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.5.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.5.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":660,"wires":[[]]},{"id":"7f277676409b85ce","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.6.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.6.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":700,"wires":[[]]},{"id":"0e11a39dd27e23b3","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.7.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.7.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":740,"wires":[[]]},{"id":"e1184fa9fc1b2e75","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.8.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.8.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":780,"wires":[[]]},{"id":"81ec8c877ccc458b","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.9.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.9.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":820,"wires":[[]]},{"id":"7473dfbd102357a2","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.10.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.10.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":860,"wires":[[]]},{"id":"be9175d1fcfc0290","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.11.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.11.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":900,"wires":[[]]},{"id":"ace6c6dbd6762da8","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.12.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.12.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":940,"wires":[[]]},{"id":"8f06f48a01d0abf3","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.13.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.13.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":980,"wires":[[]]},{"id":"0c590f30582e2e0d","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.14.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.14.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":1020,"wires":[[]]},{"id":"c27860d7b55163a4","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.15.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.15.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":1060,"wires":[[]]},{"id":"0f15762b3078c21c","type":"delay","z":"96970f5963ff2da7","name":"1s","pauseType":"delay","timeout":"2","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"allowrate":false,"outputs":1,"x":310,"y":180,"wires":[["1735ae4db805ff41","64d20491a92fa8c7"]]},{"id":"d707f3e94e148543","type":"change","z":"96970f5963ff2da7","name":"block","rules":[{"t":"set","p":"block","pt":"flow","to":"payload","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":530,"y":240,"wires":[[]]},{"id":"10e848d2fb7a5deb","type":"change","z":"96970f5963ff2da7","name":"false","rules":[{"t":"set","p":"payload","pt":"msg","to":"false","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":310,"y":380,"wires":[["d707f3e94e148543"]]},{"id":"64d20491a92fa8c7","type":"change","z":"96970f5963ff2da7","name":"true","rules":[{"t":"set","p":"payload","pt":"msg","to":"true","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":310,"y":140,"wires":[["d707f3e94e148543"]]},{"id":"07a988dfef5c46b0","type":"function","z":"96970f5963ff2da7","name":"block","func":"// variable\nlet device = {\n    block: flow.get(\"block\"),\n}\n// false\nif  (\n    device.block === false\n    )\n// \n{\n    return msg; \n}","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":850,"y":320,"wires":[["ed8bbf569aa2f8da"]]},{"id":"7e9df7b4ae3c0b53","type":"function","z":"96970f5963ff2da7","name":"block","func":"// variable\nlet device = {\n    block: flow.get(\"block\"),\n}\n// false\nif  (\n    device.block === true\n    )\n// \n{\n    return msg; \n}","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":530,"y":360,"wires":[["09f8869bf26266f4"]]},{"id":"2d18cae18e6360ba","type":"delay","z":"96970f5963ff2da7","name":"500ms","pauseType":"delay","timeout":"500","timeoutUnits":"milliseconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"allowrate":false,"outputs":1,"x":310,"y":260,"wires":[["9deefe75a941f706"]]},{"id":"825aef51dbb3b35d","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.1.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":500,"wires":[["5a65d328a2ee213c"]]},{"id":"76e57982a4fa5ad1","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.2.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":540,"wires":[["807882b3003ba754"]]},{"id":"1c20b4f9de1af513","type":"delay","z":"96970f5963ff2da7","name":"500ms","pauseType":"delay","timeout":"500","timeoutUnits":"milliseconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"allowrate":false,"outputs":1,"x":310,"y":300,"wires":[["9deefe75a941f706"]]},{"id":"d8ae5c6eb43a9514","type":"change","z":"96970f5963ff2da7","name":"Delete Data","rules":[{"t":"delete","p":"endpoint","pt":"msg"},{"t":"delete","p":"url","pt":"msg"},{"t":"delete","p":"headers","pt":"msg"},{"t":"delete","p":"method","pt":"msg"},{"t":"delete","p":"statusCode","pt":"msg"},{"t":"delete","p":"responseUrl","pt":"msg"},{"t":"delete","p":"retry","pt":"msg"},{"t":"delete","p":"redirectList","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":550,"y":440,"wires":[["07ef5070605adb2a"]]},{"id":"6a962aee9854cc9a","type":"comment","z":"96970f5963ff2da7","name":"Sets the Device Data in Flow-Variables","info":"","x":950,"y":200,"wires":[]},{"id":"b90ab8a670739d83","type":"change","z":"96970f5963ff2da7","name":"flow.device","rules":[{"t":"set","p":"device","pt":"flow","to":"payload","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":870,"y":240,"wires":[[]]},{"id":"07ef5070605adb2a","type":"split","z":"96970f5963ff2da7","name":"","splt":"\\n","spltType":"str","arraySplt":1,"arraySpltType":"len","stream":false,"addname":"","x":570,"y":760,"wires":[["23a3789b05ec994e","825aef51dbb3b35d","b48b748fa1308840","e776afa0e9806b23","f0796112e7b7b535","76e57982a4fa5ad1","4a2034902b60e4ee","a04b91018a3eabbe","6ddaf99e5edb0e70","58f7b9188864e470","197ded1bc968ce36","c4ef09023a0ad9a3","889186c9b807ea54","b7b73adea75d15e5","8e47bec12b0d34fa","2add6c6bd6a1f43d"]]},{"id":"b48b748fa1308840","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.3.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":580,"wires":[["c94227ec66636aa2"]]},{"id":"e776afa0e9806b23","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.4.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":620,"wires":[["3bd0cf0250b7a9d6"]]},{"id":"f0796112e7b7b535","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.5.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":660,"wires":[["04eaeab9bb46cf34"]]},{"id":"23a3789b05ec994e","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.0.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":460,"wires":[["b7ab41aa67eec789"]]},{"id":"b7ab41aa67eec789","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.0.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.0.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":460,"wires":[[]]},{"id":"5a65d328a2ee213c","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.1.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.1.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":500,"wires":[[]]},{"id":"807882b3003ba754","type":"change","z":"96970f5963ff2da7","name":"Name / Battery ","rules":[{"t":"set","p":"topic","pt":"msg","to":"device.2.name","tot":"flow"},{"t":"set","p":"payload.name","pt":"msg","to":"device.2.name","tot":"flow"},{"t":"move","p":"payload.power_state.battery_level","pt":"msg","to":"payload.battery_level","tot":"msg"},{"t":"delete","p":"payload.type","pt":"msg"},{"t":"delete","p":"payload.owner","pt":"msg"},{"t":"delete","p":"payload.id_v1","pt":"msg"},{"t":"delete","p":"payload.power_state","pt":"msg"},{"t":"delete","p":"payload.id","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1040,"y":540,"wires":[[]]},{"id":"4a2034902b60e4ee","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.6.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":700,"wires":[["7f277676409b85ce"]]},{"id":"a04b91018a3eabbe","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.7.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":740,"wires":[["0e11a39dd27e23b3"]]},{"id":"6ddaf99e5edb0e70","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.8.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":780,"wires":[["e1184fa9fc1b2e75"]]},{"id":"58f7b9188864e470","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.9.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":820,"wires":[["81ec8c877ccc458b"]]},{"id":"197ded1bc968ce36","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.10.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":860,"wires":[["7473dfbd102357a2"]]},{"id":"2add6c6bd6a1f43d","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.15.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":1060,"wires":[["c27860d7b55163a4"]]},{"id":"8e47bec12b0d34fa","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.14.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":1020,"wires":[["0c590f30582e2e0d"]]},{"id":"b7b73adea75d15e5","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.13.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":980,"wires":[["8f06f48a01d0abf3"]]},{"id":"889186c9b807ea54","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.12.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":940,"wires":[["ace6c6dbd6762da8"]]},{"id":"c4ef09023a0ad9a3","type":"switch","z":"96970f5963ff2da7","name":"","property":"payload.id","propertyType":"msg","rules":[{"t":"eq","v":"device.11.power","vt":"flow"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":900,"wires":[["be9175d1fcfc0290"]]},{"id":"9f608464a5f061f5","type":"comment","z":"96970f5963ff2da7","name":"valid proof","info":"","x":860,"y":420,"wires":[]},{"id":"f5b09e53a504bf40","type":"comment","z":"96970f5963ff2da7","name":"Combining Name / Battery","info":"","x":1070,"y":420,"wires":[]},{"id":"8e383874eb3e0e74","type":"comment","z":"96970f5963ff2da7","name":"http request for battery and device endpoint","info":"","x":280,"y":440,"wires":[]},{"id":"9deefe75a941f706","type":"function","z":"96970f5963ff2da7","name":"HUE API V2","func":"let api_key = env.get(\"API-Key\");\nlet ip_adress = env.get(\"IP-Adress\")\n// you can copy the user from the hue config node\nvar user = api_key;\n\n\nif(msg.endpoint === undefined)\n    msg.url = \"https://\" + ip_adress + \"/clip/v2/resource\";\nelse if(msg.topic === undefined)\n    msg.url = \"https://\" + ip_adress + \"/clip/v2/resource/\" + msg.endpoint;\nelse\n    msg.url = \"https://\" + ip_adress + \"/clip/v2/resource/\" + msg.endpoint + \"/\" + msg.topic;\n\n\nmsg.headers = {};\n\nmsg.headers['hue-application-key'] = user;\nmsg.headers['Content-Type'] = 'application/json';\nmsg.headers['Accept'] = 'application/json';\n\nif(msg.payload != undefined)\n msg.headers['ContentLength'] = JSON.stringify(msg.payload).length;\n\nif(msg.verb === undefined)\n msg.method = \"get\";\nelse\n msg.method = msg.verb;\n\n\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":550,"y":280,"wires":[["8dac534a5ce3b10d"]]},{"id":"4bc31af9bd03bbdc","type":"tls-config","name":"","cert":"","key":"","ca":"","certname":"","keyname":"","caname":"","servername":"","verifyservercert":false},{"id":"fad2db7dbef54979","type":"tab","label":"Battery Levels","disabled":false,"info":"","env":[]},{"id":"d5cc9da521feee78","type":"debug","z":"fad2db7dbef54979","name":"Battery Levels","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":820,"y":240,"wires":[]},{"id":"721af38911a98de8","type":"comment","z":"fad2db7dbef54979","name":"step1 -- Add API-Key & IP-Adress here","info":"","x":510,"y":200,"wires":[]},{"id":"f53560a09db667fe","type":"inject","z":"fad2db7dbef54979","name":"","props":[{"p":"payload"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"true","payloadType":"bool","x":250,"y":240,"wires":[["06b6599ef1eb25a1"]]},{"id":"69d4bbd3d38cd42e","type":"comment","z":"fad2db7dbef54979","name":"Payloads also have the Device Name as a topic","info":"","x":820,"y":280,"wires":[]},{"id":"3537e62fde7f6ffd","type":"comment","z":"fad2db7dbef54979","name":"step2 -- Inject","info":"","x":250,"y":200,"wires":[]},{"id":"cb783882587dbea0","type":"comment","z":"fad2db7dbef54979","name":"step3 -- wait a bit for the payloads","info":"","x":830,"y":200,"wires":[]},{"id":"06b6599ef1eb25a1","type":"subflow:96970f5963ff2da7","z":"fad2db7dbef54979","name":"","x":520,"y":240,"wires":[["d5cc9da521feee78"]]}] `

**Reply by GeorgeCastanza** [reply]
That's perfect, all my MD's are showing up with the battery - perfect, thanks!

**Reply by andesse** [reply]
@GeorgeCastanza the problem is, that the battery and the name are in different endpoints. That is so annoying. Names are in devices, batter in power. So I need to write in the other subflow the name into context and compare it with the battery endpoint data, so that it merged into name & battery level.
